src/syscall_mock/mod.rs

```diff
@@ -76,6 +76,24 @@ impl SyscallMock {
                     ptrace::setregs(pid, registers)?;
                 }
             }
+            (Syscall::Stat, SyscallStop::Exit) if !self.protocol.mocked_files.is_empty() => {
```

Comment:
Woot, woot, pattern guards. Didn't know about those.

Reply:
Although, I would remove this guard. As this is just an optimization and adds more complexity.